tests/foreman/ui/test_settings.py

```diff
@@ -175,6 +175,40 @@ def test_negative_validate_error_message(session, set_original_property_value):
                 assert is_valid_error_message(str(context.value))
 
 
+@tier2
+def test_positive_selectors(session):
+    """"Testing input for selectors: dropdown, text area, input box
+
+    :id: 529ddd3a-1271-4043-9006-eac436b08b11
+
+    :expectedresults: Successfully add value to different selectors
+
+    :CaseImportance: High
+    """
+    uuid_input = "[ {} ]".format(gen_string("alpha"))
+    random_input = gen_string('alpha')
+    uuid_duplicate_prop = ('host_dmi_uuid_duplicates', uuid_input)
+    reg_host_prop = ('register_hostname_fact', random_input)
+    dep_solve_prop = ('content_view_solve_dependencies', 'Yes')
+    #blank, blank, No
+    with session:
+        for setting_attr, value in [uuid_duplicate_prop, dep_solve_prop, reg_host_prop]:
+            # Reads and stores old value
+            old_val = session.settings.read('name = {}'.format(setting_attr))
+            # Update to new value
+            session.settings.update(
+                'name = {}'.format(setting_attr),
+                value
+            )
+            result = session.settings.read('name = {}'.format(setting_attr))
+            assert result['table'][0]['Value'] == value
+            # Resets back to old value
+            session.settings.update(
```

Comment:
Please add above line in finally block, So it will be run all of the time. Instead of using UI to update please use an API call to update that same, So it will be lesser time.

Reply:
@omkarkhatavkar, I've added the API call to reset the values taken.  There seems to be an issue with the function ```set_original_property_value``` when I tried to use it and it's not working properly in resetting the values. So instead, I used the Implementation from ```tests/foreman/api/test_setting.py::test_positive_update_hostname_prefix_without_value``` and it seems to be working fine.

Below is the test result:
```$ pytest tests/foreman/ui/test_settings.py::test_positive_selectors
========================= test session starts ===================
platform linux -- Python 3.6.8, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/ltran/Projects/robottelo
plugins: services-1.3.1, mock-1.10.4
collecting ... 2019-11-30 23:32:23 - conftest - DEBUG - Collected 1 test cases
2019-11-30 18:32:24 - robottelo.ssh - DEBUG - Instantiated Paramiko client 0x7fdaf6de1208
2019-11-30 18:32:24 - robottelo.ssh - INFO - Connected to [dell-per6415-03.lab2.eng.bos.redhat.com]
2019-11-30 18:32:24 - robottelo.ssh - INFO - >>> rpm -q satellite
2019-11-30 18:32:25 - robottelo.ssh - INFO - <<< stdout
satellite-6.7.0-4.beta.el7sat.noarch

2019-11-30 18:32:25 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7fdaf6de1208
2019-11-30 18:32:25 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2019-11-30 18:32:25 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 1 item                                                                                                                                                                                                                                                           

tests/foreman/ui/test_settings.py .                                                                                                                                                                                                                                  [100%]

============================ warnings summary ===========
tests/foreman/ui/test_settings.py::test_positive_selectors
  /home/ltran/Projects/myvenv/lib64/python3.6/site-packages/widgetastic/widget/select.py:133: DeprecationWarning: The unescape method is deprecated and will be removed in 3.5, use html.unescape() instead.
    for option

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================ 1 passed, 1 warnings in 38.74 seconds 
```